src/main/java/collections/myHashMap/Key.java

```diff
@@ -0,0 +1,24 @@
+package collections.myHashMap;
+
+import java.util.Objects;
+
+public class Key{
+    private int key;
+
+    public Key(int k) {
+        this.key = k;
+    }
+
+    @Override
+    public boolean equals(Object o){
+        if(this == o){ return true;}
+        if(o == null || getClass() != o.getClass()) return false;
+        Key key1 = (Key) o;
+        return Objects.equals(key, key1.key);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(key);
+    }
+}
```

Comment:
All in all this class is just a wrapper that holds int, and comparing two keys is basically comparing integers by i1 == i2. 
Why complicating?